src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -96,143 +103,211 @@ public class FromBase64Transform : ICryptoTransform
 
-            char[] tempChar = Encoding.ASCII.GetChars(transformBuffer, 0, 4 * numBlocks);
-            byte[] tempBytes = Convert.FromBase64CharArray(tempChar, 0, 4 * numBlocks);
-            return tempBytes;
+            return outputSize;
         }
 
-        private byte[] DiscardWhiteSpaces(byte[] inputBuffer, int inputOffset, int inputCount)
+        private void ConvertFromBase64(Span<byte> tmpBuffer, Span<byte> outputBuffer, out int consumed, out int written)
         {
-            int i, iCount = 0;
-            for (i = 0; i < inputCount; i++)
+            int bytesToTransform = _inputIndex + tmpBuffer.Length;
+
+            // Common case for bytesToTransform = 4
+            byte[] transformBufferArray = null;
+            Span<byte> transformBuffer = stackalloc byte[4];
+            if (bytesToTransform > 4)
             {
-                if (char.IsWhiteSpace((char)inputBuffer[inputOffset + i])) iCount++;
+                transformBuffer = transformBufferArray = ArrayPool<byte>.Shared.Rent(bytesToTransform);
             }
 
-            // If there's nothing to do, leave early
-            if (iCount == 0 && inputOffset == 0 &&
-                inputCount == inputBuffer.Length)
+            transformBuffer = transformBuffer.Slice(0, bytesToTransform);
+
+            // Copy _inputBuffer to transformBuffer and append tmpBuffer
+            Debug.Assert(_inputIndex < _inputBuffer.Length);
+            _inputBuffer.AsSpan(0, _inputIndex).CopyTo(transformBuffer);
+            tmpBuffer.CopyTo(transformBuffer.Slice(_inputIndex));
+
+            // Save data that won't be transformed to _inputBuffer, so it can be transformed later
+            _inputIndex = bytesToTransform & 3;     // bit hack for % 4
+            Debug.Assert(_inputIndex < _inputBuffer.Length);
+            tmpBuffer.Slice(tmpBuffer.Length - _inputIndex).CopyTo(_inputBuffer);
+
+            OperationStatus status = Base64.DecodeFromUtf8(transformBuffer, outputBuffer, out consumed, out written);
+
+            if (status == OperationStatus.InvalidData)
             {
-                return inputBuffer;
+                ThrowHelper.ThrowBase64FormatException();
             }
+            Debug.Assert(status == OperationStatus.Done);
 
-            byte[] rgbOut = new byte[inputCount - iCount];
-            iCount = 0;
-            for (i = 0; i < inputCount; i++)
+            if (transformBufferArray != null)
             {
-                if (!char.IsWhiteSpace((char)inputBuffer[inputOffset + i]))
-                {
-                    rgbOut[iCount++] = inputBuffer[inputOffset + i];
-                }
+                transformBuffer.Clear();
+                ArrayPool<byte>.Shared.Return(transformBufferArray);
             }
-
-            return rgbOut;
         }
 
-        private static void ValidateTransformBlock(byte[] inputBuffer, int inputOffset, int inputCount)
+        public void Clear()
```

Comment:
On this class, Clear is documented as calling Dispose.  So please restore the call graph.

https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.frombase64transform.clear?view=netframework-4.8